### PR TITLE
CrowdstrikeFalcon - Fixed 404 error message for IOC's 

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/31450

## Screenshots
image

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 1]
```diff
@@ -162,7 +162,11 @@ def http_request(method, url_suffix, params=None, data=None, files=None, headers
         :param get_token_flag: If set to True will call get_token()
 
         :type no_json: ``bool``
-        :param no_json: If set to true will not parse the content and will return the raw response object for successful response
+        :param no_json: If set to true will not parse the content and will return the raw response object for successful
+        response
+
+        :type status_code: ``int``
+        :param: status_code: The request codes to accept as OK.
```
Comment (ShahafBenYakir) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:169]:
```suggestion
        :param no_json: If set to true will not parse the content and will return the raw response object for successful
        response
        :type status_code: ``int``
        :param: status_code: The request codes to accept as OK.
```

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 2]
```diff
@@ -1463,6 +1471,10 @@ def get_ioc_device_count_command(ioc_type: str, value: str):
     :param value: The IOC value
     """
     raw_res = get_ioc_device_count(ioc_type, value)
+    errors = raw_res.get('errors', [])
+    for error in errors:
+        if error.get('code') == 404:
+            return f'No results found for {ioc_type} - {value}'
```
Comment (ShahafBenYakir) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:1477]: I think this should go to the `get_ioc_device_count` function so you could pass to the `http_request` the `status_code arg`

**Review by ShahafBenYakir — CHANGES_REQUESTED** [review]

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py (hunk 1 above)
Comment (Shellyber) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:169]: What's the differnece here?

Comment (ShahafBenYakir) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:169]: Just a space

Comment (Shellyber) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:169]: looks ok in the code itself. probably git glich
image

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 3]
```diff
@@ -1463,14 +1476,17 @@ def get_ioc_device_count_command(ioc_type: str, value: str):
...
+        handle_response_errors(raw_res)
+        device_count_res = raw_res.get('resources')
+        ioc_id = f"{ioc_type}:{value}"
+        if not device_count_res:
+            return create_entry_object(raw_res, hr=f"Could not find any devices the IOC **{ioc_id}** was detected in.")
+        context = [get_trasnformed_dict(device_count, IOC_DEVICE_COUNT_MAP) for device_count in device_count_res]
+        hr = f'Indicator of Compromise **{ioc_id}** device count: **{device_count_res[0].get("device_count")}**'
+        return create_entry_object(contents=raw_res, ec={'CrowdStrike.IOC(val.ID === obj.ID)': context}, hr=hr)
+    else:
+        return raw_res
```
Comment (ShahafBenYakir) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:1489]:
```suggestion
    if 'No results found for' in raw_res:
        return raw_res
    else:
        handle_response_errors(raw_res)
        device_count_res = raw_res.get('resources')
        ioc_id = f"{ioc_type}:{value}"
        if not device_count_res:
            return create_entry_object(raw_res, hr=f"Could not find any devices the IOC **{ioc_id}** was detected in.")
        context = [get_trasnformed_dict(device_count, IOC_DEVICE_COUNT_MAP) for device_count in device_count_res]
        hr = f'Indicator of Compromise **{ioc_id}** device count: **{device_count_res[0].get("device_count")}**'
        return create_entry_object(contents=raw_res, ec={'CrowdStrike.IOC(val.ID === obj.ID)': context}, hr=hr)
```

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**